server/hotrod/src/main/scala/org/infinispan/server/hotrod/AbstractEncoder1x.scala

```diff
@@ -165,6 +165,10 @@ abstract class AbstractEncoder1x extends AbstractVersionedEncoder with Constants
       val cache = server.getCacheInstance(r.cacheName, addressCache.getCacheManager, false)
       val cacheMembers = cache.getRpcManager.getMembers
       val serverEndpointsMap = addressCache.toMap
+
+      if (serverEndpointsMap.isEmpty)
+         logServerEndpointTopologyEmpty(cacheMembers.toString)
```

Comment:
This check overlaps with the other serverEndpointsMap check below, maybe it would be cleared if you only logged the message when the topology update is sent?